src/main/drivers/rx/rx_sx127x.h

```diff
@@ -0,0 +1,338 @@
+#define SX127x_TXCONTINUOUSMODE_ON 0x08
+#define SX127x_TXCONTINUOUSMODE_OFF 0x00
+#define SX127x_PPMOFFSET 0x27
+
+///// SX1278 Regs /////
+//SX1278 specific register map
+#define SX1278_REG_MODEM_CONFIG_3 0x26
+#define SX1278_REG_TCXO 0x4B
+#define SX1278_REG_PA_DAC 0x4D
+#define SX1278_REG_FORMER_TEMP 0x5D
+#define SX1278_REG_AGC_REF 0x61
+#define SX1278_REG_AGC_THRESH_1 0x62
+#define SX1278_REG_AGC_THRESH_2 0x63
+#define SX1278_REG_AGC_THRESH_3 0x64
+#define SX1278_REG_PLL 0x70
+
+//SX1278 LoRa modem settings
+//SX1278_REG_OP_MODE                                                  MSB   LSB   DESCRIPTION
+#define SX1278_HIGH_FREQ 0x00 //  3     3     access HF test registers
+#define SX1278_LOW_FREQ 0x08  //  3     3     access LF test registers
+
+//SX1278_REG_FRF_MSB + REG_FRF_MID + REG_FRF_LSB
+#define SX1278_FRF_MSB 0x6C //  7     0     carrier frequency setting: f_RF = (F(XOSC) * FRF)/2^19
+#define SX1278_FRF_MID 0x80 //  7     0         where F(XOSC) = 32 MHz
+#define SX1278_FRF_LSB 0x00 //  7     0               FRF = 3 byte value of FRF registers
+
+//SX1278_REG_PA_CONFIG
+#define SX1278_MAX_POWER 0x70 //  6     4     max power: P_max = 10.8 + 0.6*MAX_POWER [dBm]; P_max(MAX_POWER = 0b111) = 15 dBm
+//#define SX1278_MAX_POWER                              0x10  //  6     4     changed
+
+//SX1278_REG_LNA
+#define SX1278_LNA_BOOST_LF_OFF 0x00 //  4     3     default LNA current
+
+#define SX1278_HEADER_EXPL_MODE 0x00 //  0     0     explicit header mode
+#define SX1278_HEADER_IMPL_MODE 0x01 //  0     0     implicit header mode
+
+//SX1278_REG_MODEM_CONFIG_2
+#define SX1278_RX_CRC_MODE_OFF 0x00 //  2     2     CRC disabled
+#define SX1278_RX_CRC_MODE_ON 0x04  //  2     2     CRC enabled
+
+//SX1278_REG_MODEM_CONFIG_3
+#define SX1278_LOW_DATA_RATE_OPT_OFF 0x00 //  3     3     low data rate optimization disabled
+#define SX1278_LOW_DATA_RATE_OPT_ON 0x08  //  3     3     low data rate optimization enabled
+#define SX1278_AGC_AUTO_OFF 0x00          //  2     2     LNA gain set by REG_LNA
+#define SX1278_AGC_AUTO_ON 0x04           //  2     2     LNA gain set by internal AGC loop
+
+#define SX1276_HEADER_EXPL_MODE 0x00 //  0     0     explicit header mode
+#define SX1276_HEADER_IMPL_MODE 0x01 //  0     0     implicit header mode
+
+#define SX127x_ERR_NONE 0x00
+#define SX127x_ERR_CHIP_NOT_FOUND 0x01
+#define SX127x_ERR_EEPROM_NOT_INITIALIZED 0x02
+
+#define SX127x_ERR_PACKET_TOO_LONG 0x10
+#define SX127x_ERR_TX_TIMEOUT 0x11
+
+#define SX127x_ERR_RX_TIMEOUT 0x20
+#define SX127x_ERR_CRC_MISMATCH 0x21
+
+#define SX127x_ERR_INVALID_BANDWIDTH 0x30
+#define SX127x_ERR_INVALID_SPREADING_FACTOR 0x31
+#define SX127x_ERR_INVALID_CODING_RATE 0x32
+#define SX127x_ERR_INVALID_FREQUENCY 0x33
+
+#define SX127x_ERR_INVALID_BIT_RANGE 0x40
+
+#define SX127x_CHANNEL_FREE 0x50
+#define SX127x_PREAMBLE_DETECTED 0x51
+
+#define SX127x_SPI_READ 0x00
+#define SX127x_SPI_WRITE 0x80
+
+#define SX127x_MAX_POWER 0x0F //17dBm
+
+#define SX127x_FREQ_STEP 61.03515625
```

Comment:
the problem is - it needs to be consistent with the original firmware, otherwise there might be a slight misalignment between frequencies leading to decreased range. I'll keep it.